### DISCUSSION TITLE
Vertical spacing locked

### DISCUSSION THREAD

**Discussion by elramalho** [joaomlourenco/novathesis#375] [Q&A]
Hello,

Writing a master thesis with University of Minho template.

If I have a big picture before and after some text, the text is spread along the page, which makes the space between the paragraphs bigger.

Is there any way to prevent this? 
I prefer to have the text with a normal space between paragraphs and then a big space of text at the end of the page.

<img width="612" alt="Screenshot 2022-10-20 at 17 47 46" src="https://user-images.githubusercontent.com/116292986/197009780-f92c5f97-5542-4215-9c52-75ad83cb2bf2.png">

**Comment by b-pereira** [comment]
Sorry, I didn't quite understand your question. 

Do you want to set the images on those exact positions? 

If you want to do that, it's possible. However it's not recommended - images are floats, i.e., you should let LaTeX decide where it's an excellent place to position your images and reference the float (it could also be a table,  a code listing, etc.).**That's good typography.** The other way around, the document will have too much blank space.

Notwithstanding, it's better to show the LaTeX code used to produce the PDF page you have shown. You can redact what you don't want to show, like figure input name or text.

**Reply by joaomlourenco** [reply]
Following @b-pereira just use, just use
```latex
\begin{figure}[htp]
…
\end{figure}
```
and everything will look as it should (according to the typography rules that we, mere mortals, totally ignore and that LaTeX knows by heart).
